src/Carousel.js

```diff
@@ -301,4 +303,8 @@ const Carousel = React.createClass({
   }
 });
 
+Carousel = Object.assign(Carousel, { Caption });
+
 export default Carousel;
+
+export { Caption };
```

Comment:
The downside is a breaking change, the upside is consistency with e.g. Media.

That being said, I defer to you. Let me know which you'd like me to do, PR to 0.29 branch or modify the code.